usr.bin/procstat/procstat_compartments.c

```diff
@@ -50,14 +50,13 @@ procstat_compartments(struct procstat *procstat, struct kinfo_proc *kipp)
 		xo_emit("{T:/%5s %-19s %4s %-40s}\n", "PID", "COMM", "CID",
 		    "CNAME");
 	if (procstat_getcompartments(procstat, kipp, &cccp, &ncomparts) != 0)
-		goto out;
+		return;
```

Comment:
I think this is the only real fix FWIW.  It's not unusual to say that the value of arguments passed by pointer is unchanged/undefined if a function fails.  Or put another way, it is only assigned a defined value if the function succeeds and it is an error to use the value if the function fails.

Reply:
Sounds good to me to just leave the inout pointers untouched if there's an error. The diff is also smaller this way.